eclair-core/src/main/scala/fr/acinq/eclair/db/PaymentsDb.scala

```diff
@@ -19,70 +19,131 @@ package fr.acinq.eclair.db
 import java.util.UUID
 
 import fr.acinq.bitcoin.ByteVector32
-import fr.acinq.eclair.MilliSatoshi
-import fr.acinq.eclair.payment.PaymentRequest
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.eclair.payment._
+import fr.acinq.eclair.router.Hop
+import fr.acinq.eclair.{MilliSatoshi, ShortChannelId}
 
 trait PaymentsDb {
 
-  // creates a record for a non yet finalized outgoing payment
+  /** Create a record for a non yet finalized outgoing payment. */
   def addOutgoingPayment(outgoingPayment: OutgoingPayment)
 
-  // updates the status of the payment, if the newStatus is SUCCEEDED you must supply a preimage
-  def updateOutgoingPayment(id: UUID, newStatus: OutgoingPaymentStatus.Value, preimage: Option[ByteVector32] = None)
+  /** Update the status of the payment in case of success. */
+  def updateOutgoingPayment(paymentResult: PaymentSent)
```

Comment:
Nit: could have been an `Either[PaymentFailed, PaymentSent]`. Not sure what's best

Reply:
Not a strong opinion, but I slightly prefer overloading in that case (it's easier for the caller).